### PR TITLE
[ONNX] Update docs on symbolic registration

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85350
* #85349
* #85312
* #85165
* #84776
* #84728
* __->__ #85290
* #84448

- Move inline instructions on editing symbolic functions to the README
- Add a line on using the symbolic function registration decorator.

### CODE REVIEW

**Comment by justinchuby** [comment]
@BowenBao PTAL

File: torch/onnx/README.md [hunk 1]
```diff
@@ -23,3 +24,75 @@ symbolic_opset9.py.
...
+some moving parts to implementing the ONNX translation in this case:
+
+- By the time we get the scalar in a symbolic function here, it is no longer
+    a Python long/float, but a PyTorch tensor with `numel == 1` (eventually, we
+    want it to be a zero dim tensor but this change has not happened yet.)
+    However, the type of this scalar is *exactly* what the user wrote in
+    Python, which may not match the tensor it is being added to.  PyTorch
+    will do implicit conversions on scalars; however, ONNX will not, so
+    we must do the conversion ourselves.  This is what `symbolic_helper._if_scalar_type_as()`
+    does.
```
Comment (BowenBao) [inline torch/onnx/README.md:85]: Noticed this is a little outdated, best add
```suggestion
    and `_jit_pass_onnx_scalar_type_analysis` does.
```

**Review by BowenBao — APPROVED** [review]

File: torch/onnx/README.md (hunk 1 above)
Comment (justinchuby) [reply torch/onnx/README.md:85]: Done

**Pull request closed without merging**